activesupport/lib/active_support/deprecation.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require "singleton"
+require "concurrent/utility/engine"
 require "concurrent/atomic/thread_local_var"
```

Comment:
I see the NME, but apparently "concurrent/utility/*" should be required in the concurrent-ruby internally, so I think it is an issue in the concurrent-ruby (I filed an issue on the concurrent-ruby https://github.com/ruby-concurrency/concurrent-ruby/pull/833).

Until this is fixed in the concurrent-ruby, how about just reordering require to address the NME?

```diff
diff --git a/activesupport/lib/active_support/deprecation.rb b/activesupport/lib/active_support/deprecation.rb
index 414bd08e06..813f6fefe2 100644
--- a/activesupport/lib/active_support/deprecation.rb
+++ b/activesupport/lib/active_support/deprecation.rb
@@ -1,7 +1,6 @@
 # frozen_string_literal: true

 require "singleton"
-require "concurrent/atomic/thread_local_var"

 module ActiveSupport
   # \Deprecation specifies the API used by Rails to deprecate methods, instance
@@ -22,6 +21,7 @@ class Deprecation
     require "active_support/deprecation/method_wrappers"
     require "active_support/deprecation/proxy_wrappers"
     require "active_support/core_ext/module/deprecation"
+    require "concurrent/atomic/thread_local_var"

     include Singleton
     include InstanceDelegator
```

Reply:
Good call, looks like that solves it. I'll correct my PR.